demo_spaceros/Dockerfile

```diff
@@ -0,0 +1,106 @@
+# Copyright 2021 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# A Docker configuration script to build the Space ROS image.
+#
+# The script provides the following build arguments:
+#
+#   VCS_REF     - The git revision of the Space ROS source code (no default value).
+#   VERSION     - The version of Space ROS (default: "preview")
+#   WORKSPACE   - The location for the Space ROS source code in the container (default: /root/src/spaceros_ws)
+
+FROM openrobotics/spaceros:latest
+
+ARG VCS_REF
+ARG VERSION="preview"
+ARG WORKSPACE=/root/src/spaceros_demo_ws
+
+WORKDIR ${WORKSPACE}
+
+# LABEL the image
+LABEL org.label-schema.schema-version="1.0"
+LABEL org.label-schema.name="Space ROS"
+LABEL org.label-schema.description="Preview demo of the Space ROS platform"
+LABEL org.label-schema.vendor="Open Robotics"
+LABEL org.label-schema.version=${VERSION}
+LABEL org.label-schema.url="https://github.com/space-ros"
+LABEL org.label-schema.vcs-url="https://github.com/space-ros/docker-images"
+LABEL org.label-schema.vcs-ref=${VCS_REF}
+
+# Disable prompting during package installation
+ARG DEBIAN_FRONTEND=noninteractive
+
+# Make sure the latest versions of packages are installed
+RUN apt-get update
+RUN apt-get dist-upgrade -y
```

Comment:
Torn between saying this is something that the base image should handle and knowing that I just put it in our recommended practices for installation. However, this creates additional layers in the Dockerfile and there is no reliable way for these updates to be cache-cleared when apt sources change. So I think that having this in "downstream" containers creates false confidence that things are properly up-to-date.